src/realm.h

```diff
@@ -874,6 +887,13 @@ RLM_API bool realm_close(realm_t*);
  */
 RLM_API bool realm_is_closed(realm_t*);
 
+/**
+ * Begin a read transaction for the Realm file.
+ *
+ * @return True if no exception occurred.
+ */
+RLM_API bool realm_begin_read(realm_t*);
```

Comment:
Under what conditions can this function be called? The Realm is implicitly in a read transaction, so do we need a separate `realm_end_read()` call?

Reply:
The ObjectStore `SharedRealm` isn't, it doesn't start it until you interact with it. See https://mongodb.slack.com/archives/C010R3CMP3N/p1618494638175200

For our use case, we don't need `realm_end_read()`. At least right now.